### PR TITLE
Update parent to version version 13

### PR DESCRIPTION
**Pull request by cescoffier**
Supersedes #260

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -231,10 +231,17 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-javadoc-plugin</artifactId>
+          <version>3.1.0</version>
```
Comment (kenfinnigan) [inline pom.xml]: Do these versions not come from the parent?

If not, we should fix that

Comment (cescoffier) [reply pom.xml]: I pinned it because I want to have the control of it. It's mostly to support new version of java with the annoying linter.

Comment (kenfinnigan) [reply pom.xml]: so you don't think we should be managing these versions in a parent, for any SmallRye project?

Comment (cescoffier) [reply pom.xml]: Both :-)

The parent should give a recent enough version.
Projects can decide to override to tweak their build.

Comment (kenfinnigan) [reply pom.xml]: fair enough

**Pull request merged**